metrics/innodb.buffer-pool/innodb.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2024 Block, Inc.
+
+package innodbbufferpool
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/cashapp/blip"
+	"github.com/cashapp/blip/sqlutil"
+)
+
+// Table collects buffer pool data from information_schema.innodb_buffer_pool_stats.
+// https://dev.mysql.com/doc/refman/8.4/en/information-schema-innodb-buffer-pool-stats-table.html
+
+const (
+	DOMAIN  = "innodb.buffer-pool"
+	OPT_ALL = "all"
+
+	BASE_QUERY = "SELECT %s FROM information_schema.innodb_buffer_pool_stats"
+)
+
+var (
+	columnNames = []string{
+		"pool_size",
+		"free_buffers",
+		"database_pages",
+		"old_database_pages",
+		"modified_database_pages",
+		"pending_decompress",
+		"pending_reads",
+		"pending_flush_lru",
+		"pending_flush_list",
+		"pages_made_young",
+		"pages_not_made_young",
+		"pages_made_young_rate",
+		"pages_made_not_young_rate",
+		"number_pages_read",
+		"number_pages_created",
+		"number_pages_written",
+		"pages_read_rate",
+		"pages_create_rate",
+		"pages_written_rate",
+		"number_pages_get",
+		"hit_rate",
+		"young_make_per_thousand_get",
+		"not_young_make_per_thousand_get",
+		"number_pages_read_ahead",
+		"number_read_ahead_evicted",
+		"read_ahead_rate",
+		"read_ahead_evicted_rate",
+		"lru_io_total",
+		"lru_io_current",
+		"uncompress_total",
+		"uncompress_current",
+	}
+
+	columnSum map[string]string
+)
+
+func init() {
+	columnSum = make(map[string]string, len(columnNames))
+	for _, name := range columnNames {
+		columnSum[name] = fmt.Sprintf("SUM(%s) as %s", name, name)
+	}
+}
+
+// BufferPoolStats collections from the information_schema.innodb_buffer_pool_stats table.
+type BufferPoolStats struct {
+	db    *sql.DB
+	query map[string]string
+}
+
+// Verify collector implements blip.Collector interface.
+var _ blip.Collector = &BufferPoolStats{}
+
+// NewTable makes a new Table collector,
+func NewBufferPoolStats(db *sql.DB) *BufferPoolStats {
+	return &BufferPoolStats{
+		db:    db,
+		query: map[string]string{},
+	}
+}
+
+// Domain returns the Blip metric domain name (DOMAIN const).
+func (t *BufferPoolStats) Domain() string {
+	return DOMAIN
+}
+
+// Help returns the output for blip --print-domains.
+func (t BufferPoolStats) Help() blip.CollectorHelp {
+	return blip.CollectorHelp{
+		Domain:      DOMAIN,
+		Description: "Buffer Pool Stats (summed over all pools)",
+		Options: map[string]blip.CollectorHelpOption{
+			OPT_ALL: {
+				Name:    OPT_ALL,
+				Desc:    "Collect all metrics",
```

Comment:
Heh a little better description please :-)

Reply:
This is the exact same description for this option as we have in the `innodb` domain...but I'll update it.